packages/clay-card/src/index.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Context from './Context';
+import {CardDisplayType} from './types';
+
+import {AspectRatio, CardAspectRatioType} from './AspectRatio';
+import {Body, CardBodyProps, CardBodyType} from './Body';
+import {CardDescriptionType, Description} from './Description';
+import {CardDetailType, Detail} from './Detail';
+import {CardGroupType, Group} from './Group';
+import {CardHeaderType, Header} from './Header';
+
+export interface CardProps {
+	displayType?: CardDisplayType;
+
+	/**
+	 * Flag that indicates if the card can be selectable.
+	 */
+	selectable?: boolean;
+}
+
+interface Props
+	extends CardProps,
+		React.BaseHTMLAttributes<HTMLAnchorElement | HTMLDivElement> {
+	children:
+		| React.ReactElement<CardBodyProps>[]
+		| React.ReactElement<CardBodyProps>;
+}
+
+const ClayCard: React.FunctionComponent<Props> & {
+	AspectRatio: CardAspectRatioType;
+	Body: CardBodyType;
+	Detail: CardDetailType;
+	Description: CardDescriptionType;
+	Group: CardGroupType;
+	Header: CardHeaderType;
+} = ({
+	children,
+	className,
+	displayType = 'directory',
+	selectable = false,
+	...otherProps
+}) => {
+	const Content: React.FunctionComponent<Props> = ({children}) => (
```

Comment:
I know we talked about this here, but maybe we should move it out of the component. What do you think @bryceosterhaus. I would not mind keeping it here or move but it would be nice to have a practice for it.

Reply:
Its probably fine to leave here, or we could try and inline it into the render return as well.

Reply:
I guess that we can leave here, once this component is only used in this file and I don't have plans to expose a context provider of `displayType` to users(?)